File: articles/social-norms-and-institutional-cooperation/go/social_norms_score.go
package main

import "fmt"

type SocialNormsCase struct {
	DescriptiveNorm          float64
	InjunctiveNorm           float64
	Trust                    float64
	Legitimacy               float64
	SanctionIntensity        float64
	TransmissionStrength     float64
	InstitutionalReinforcement float64
	NormConflict             float64
	HypocrisyVisibility      float64
	UnequalEnforcement       float64
	PerformativeCompliance   float64
	DistributionalAttention  float64
}

func SocialNormsCooperationScoreRaw(x SocialNormsCase) float64 {
	return 0.14*x.DescriptiveNorm +
		0.14*x.InjunctiveNorm +
		0.13*x.Trust +
		0.12*x.Legitimacy +
		0.10*x.SanctionIntensity +
		0.11*x.TransmissionStrength +
		0.12*x.InstitutionalReinforcement -
		0.13*x.NormConflict -
		0.08*x.HypocrisyVisibility -
		0.07*x.UnequalEnforcement -
		0.05*x.PerformativeCompliance +
		0.04*x.DistributionalAttention
}

func main() {
	demo := SocialNormsCase{80, 78, 75, 72, 60, 74, 82, 30, 25, 20, 25, 70}
	fmt.Printf("Social norms cooperation raw score: %.2f\n", SocialNormsCooperationScoreRaw(demo))
}

File: articles/social-norms-and-institutional-cooperation/rust/social_norms_score.rs
struct SocialNormsCase {
    descriptive_norm: f64,
    injunctive_norm: f64,
    trust: f64,
    legitimacy: f64,
    sanction_intensity: f64,
    transmission_strength: f64,
    institutional_reinforcement: f64,
    norm_conflict: f64,
    hypocrisy_visibility: f64,
    unequal_enforcement: f64,
    performative_compliance: f64,
    distributional_attention: f64,
}

fn social_norms_cooperation_score_raw(x: &SocialNormsCase) -> f64 {
    0.14 * x.descriptive_norm
        + 0.14 * x.injunctive_norm
        + 0.13 * x.trust
        + 0.12 * x.legitimacy
        + 0.10 * x.sanction_intensity
        + 0.11 * x.transmission_strength
        + 0.12 * x.institutional_reinforcement
        - 0.13 * x.norm_conflict
        - 0.08 * x.hypocrisy_visibility
        - 0.07 * x.unequal_enforcement
        - 0.05 * x.performative_compliance
        + 0.04 * x.distributional_attention
}

fn main() {
    let demo = SocialNormsCase {
        descriptive_norm: 80.0,
        injunctive_norm: 78.0,
        trust: 75.0,
        legitimacy: 72.0,
        sanction_intensity: 60.0,
        transmission_strength: 74.0,
        institutional_reinforcement: 82.0,
        norm_conflict: 30.0,
        hypocrisy_visibility: 25.0,
        unequal_enforcement: 20.0,
        performative_compliance: 25.0,
        distributional_attention: 70.0,
    };

    println!(
        "Social norms cooperation raw score: {:.2}",
        social_norms_cooperation_score_raw(&demo)
    );
}
